python/run-tests.py

```diff
@@ -160,11 +160,15 @@ def run_individual_python_test(target_dir, test_name, pyspark_python):
 
 
 def get_default_python_executables():
-    python_execs = [x for x in ["python2.7", "python3.6", "pypy"] if which(x)]
-    if "python2.7" not in python_execs:
-        LOGGER.warning("Not testing against `python2.7` because it could not be found; falling"
-                       " back to `python` instead")
-        python_execs.insert(0, "python")
+    python_execs = [x for x in ["python3.6", "pypy3"] if which(x)]
+    if "python3.6" not in python_execs:
+        if which("python"):
+            LOGGER.warning("Not testing against `python3.6` because it could not be found; falling"
+                           " back to `python` instead")
+            python_execs.insert(0, "python")
```

Comment:
Can this fall back to python 2 actually?

Reply:
well, if we're not officially supporting it, should we test against it?

2.4 will still test against py2.7...

Reply:
ah, my wording could be misleading. I meant this could possibly fall back to python 2 accidentally because it falls back to test against `python`?

Reply:
sure, i *could* do that, but to what end?  

what version of python is installed by default?  

what if that version (at this point in time typically 2.7.x) isn't supported by spark 3.x+?

Reply:
ur, I actually want to say is, maybe we should fall back by:

```
if which("python3"):
    ...
    python_execs.insert(0, "python3")` 
```

instead of `python_execs.insert(0, "python")`?

Because python does more likely point to python 2.x version?
Maybe this is not a problem.

Reply:
Hm, let's leave it as is for now in this PR. I think it's controversial enough but we have multiple places where it falls back to `python`.

Reply:
> ur, I actually want to say is, maybe we should fall back by:
> 
> ```
> if which("python3"):
>     ...
>     python_execs.insert(0, "python3")` 
> ```
> 
> instead of `python_execs.insert(0, "python")`?
> 
> Because python does more likely point to python 2.x version?
> Maybe this is not a problem.

aaaaaaah ok, yeah.  i see what you mean.  yes, let me do that now.  this is the right thing to do.

Reply:
take a look at my most recent changes and let me know what you think!  everything is passing, and i've tested the code against a system w/o python3 in `PATH`...